File: leetcode/level1/leet509.js
/**
 * @param {number} N
 * @return {number}
 */
var fib = function(N) {
    let dp = []
    for (let index=0; index <= N; index++) {
        // console.log(index);
        dp[index] = index < 2 ? index : dp[index-1] + dp[index -2]
    }
    const res = dp.pop()
    console.log(res);
    return res
};

var testData = [
    2,
    3,
    4,
]

for (let item of testData) {
    // console.log(item);
    fib(item)
}

File: leetcode/level1/leet509.py
# py2
# fibonacci-number 

class Solution(object):
  def fib(self, N):
    """
    :type N: int
    :rtype: int
    """
    if N < 1:
      return 0
    elif N < 3:
      return 1
    dp = [None for i in range(N+1)]
    dp[0] = 0
    [dp[1], dp[2]] = [1, 1]
    i = 3
    while i <= N:
      dp[i] = dp[i-1] + dp[i-2]
      i += 1
    return dp[-1]


if __name__ == "__main__":
  testData = [0, 2, 3, 4, 5, 6]
  test = Solution()
  for i in testData:
    print test.fib(i)
  print test.fib(5)
  